### ISSUE TITLE
Move //mojo/services/.../public into //mojo/public/services/...

### ISSUE THREAD

**Issue by j9brown** [domokit/mojo#443]
Consider clustering the public contracts defined in the Mojo repository in one place instead of scattering them around.  Note that the private implementations of these services should remain separate, especially since they contain OS dependencies.

**Comment by viettrungluu-cr** [comment]
The reason behind the current organization is that //mojo/public contains
the core API, with things that everyone agrees on.

//mojo/services just happens to be whatever stuff random services consider
to be public (some of which may be or become standardized at some point).

We might be happier having //service/foo/public for foo's public stuff, but
currently there's no way for foo to "publish" its public stuff.

**Comment by j9brown** [comment]
Yeah.  I'm sort of torn on whether the contracts should like next to the
implementation at all.  I'm inclined to say no for code hygiene reasons,
especially since we have executables which implement multiple interfaces.
It gets a bit confusing when "services" is used to refer both to group a
bunch of related interfaces in the source tree and used to refer to some
embodiment as an app (or daemon), of which there might be several.

//mojo/public/core
//mojo/public/interfaces
//apps

Jeff.
